tests/samples_test.py

```diff
@@ -0,0 +1,359 @@
+        * numpy.array(
+            [
+                0,
+                1 / 3,
+                2 / 3,
+                1 / 9,
+                2 / 9,
+                4 / 9,
+                5 / 9,
+                7 / 9,
+                8 / 9,
+            ]
+        ),
+    ),
+]
+
+sample_points_ids = [
+    "ClenshawCurtis [0]",
+    "ClenshawCurtis [1]",
+    "ClenshawCurtis [2]",
+    "ClenshawCurtis [3]",
+    "ClenshawCurtis [8]",
+    "Trigonometric [0]",
+    "Trigonometric [1]",
+    "Trigonometric [2]",
+    "Trigonometric [3]",
+    "Trigonometric [9]",
+    "NestedClenshawCurtis [0]",
+    "NestedClenshawCurtis [1]",
+    "NestedClenshawCurtis [2]",
+    "NestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [0]",
+    "SlowNestedClenshawCurtis [1]",
+    "SlowNestedClenshawCurtis [2]",
+    "SlowNestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [4]",
+    "SlowNestedClenshawCurtis [5]",
+    "NestedTrigonometric [0]",
+    "NestedTrigonometric [1]",
+    "NestedTrigonometric [2]",
+]
+
+nested_sample_ids = [
+    "NestedClenshawCurtis [0]",
+    "NestedClenshawCurtis [1]",
+    "NestedClenshawCurtis [2]",
+    "NestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [0]",
+    "SlowNestedClenshawCurtis [1]",
+    "SlowNestedClenshawCurtis [2]",
+    "SlowNestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [4]",
+    "SlowNestedClenshawCurtis [5]",
+    "NestedTrigonometric [0]",
+    "NestedTrigonometric [1]",
+    "NestedTrigonometric [2]",
+]
+
+
+def test_clenshaw_initial():
+    """test default properties"""
+    f = smolyay.samples.ClenshawCurtisPointSet([-2, 1], 7)
+    assert numpy.array_equal(f.domain, [-2, 1])
+    assert f.degree == 7
+    assert isinstance(f.degree, int)
+    f.degree = float(5)
+    assert f.degree == 5
+    assert isinstance(f.degree, int)
+
+
+def test_trig_initial():
+    """test default properties"""
+    f = smolyay.samples.TrigonometricPointSet([0, 4 * numpy.pi], 7)
+    assert numpy.array_equal(f.domain, [0, 4 * numpy.pi])
+    assert f.frequency == 7
+    assert isinstance(f.frequency, int)
+    f.frequency = float(5)
+    assert f.frequency == 5
+    assert isinstance(f.frequency, int)
+
+
+@pytest.mark.parametrize(
+    "nested_samples",
+    [
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_nested_initial(nested_samples):
+    """test default properties"""
+    f = nested_samples([-10, 10], 7)
+    assert numpy.array_equal(f.domain, [-10, 10])
+    assert f.max_level == 7
+    assert isinstance(f.max_level, int)
+    f.max_level = float(5)
+    assert f.max_level == 5
+    assert isinstance(f.max_level, int)
+
+
+@pytest.mark.parametrize(
+    "samples",
+    [
+        smolyay.samples.ClenshawCurtisPointSet,
```

Comment:
I suggest you pass the arguments needed to make the set as an extra parameter like this, then unpack them as keyword args below. That way, you don't need to rely on all the sets having exactly the same constructor signature. I had to read this a couple times to figure out how it worked.
```suggestion
        (smolyay.samples.ClenshawCurtisPointSet, {"degree": 7})
```